### ISSUE TITLE
Better fog API structure (visible, colours etc)

### ISSUE THREAD

**Issue by Zughy** [minetest/minetest#14176]
### Problem

Currently, MT fog is quite limited, with people in these years requesting features like colour (#7284) and tweakable cloud fog (#7838). Also, players can turn it off by just pressing F3, breaking whatever aim the modder had (#14163)

### Solutions

First of all, I suggest a `visible` parameter, both to create coherence with other celestial vault elements (sun, moon, stars and clouds) and to avoid this `-1` hack on `fog_distance`, `fog_start` and whatever parameter may come next.  I maintain a mod that includes an in-game editor for the celestial vault, and implementing fog hasn't been fun as I had to treat these parameters as a special case. Furthermore, it can be a nice way to tackle #14163, enforcing `visible` when declared (no matter if `true` or `false`)

Then I'd move `fog_sun_tint`, `fog_moon_tint` and `fog_tint_type` into `fog`  

Last but not least, I wonder why fog is in `set_sky`. Like, in my mind is a weather condition, not a component of the celestial vault. Wouldn't be better to have a separate `set_fog()` function?

### Alternatives

---

**Comment by lhofhansl** [comment]
I agree with most of your suggestions. Where to put it in the API is always hard. Is it where you set clouds? Is it where you set lighting?

**Comment by Zughy** [comment]
Definitely not lighting, it's not a light phenomenon. It's humidity coming from the soil, so it doesn't belong to the celestial vault either. In my mind it's something on its own, related to visibility

**Comment by kromka-chleba** [comment]
Btw fog is not displayed on the sky if a skybox is defined, so any improvements should take this into account.

![no_fog](https://github.com/minetest/minetest/assets/65868699/3e30dcd7-1b0e-4022-899d-8cd5bdab3556)

**Comment by kromka-chleba** [comment]
>It's humidity coming from the soil, so it doesn't belong to the celestial vault either. In my mind it's something on its own, related to visibility

Or smog, dust, etc.
https://en.wikipedia.org/wiki/Aerosol

**Comment by appgurueu** [comment]
`set_fog` makes sense to me. `get_fog` should probably return a table of fog parameters if fog is set, or `nil` if fog is disabled. `set_fog()` / `set_fog(nil)` would disable fog.

**Comment by kromka-chleba** [comment]
Would be nice if you could plug a 3D noise texture into the fog to make a cool volumetric effect.

**Comment by Zughy** [comment]
This is not doable before the next major release. See https://github.com/luanti-org/luanti/pull/16725#issuecomment-3622450532
